src/storage.ts

```diff
@@ -109,16 +105,20 @@ export class Storage {
   }
 
   /**
-   * Download and return the path to a working binary using the provided client.
+   * Download and return the path to a working binary for the deployment with
+   * the provided label using the provided client.  If the label is empty, use
+   * the old deployment-unaware path instead.
+   *
    * If there is already a working binary and it matches the server version,
    * return that, skipping the download.  If it does not match but downloads are
    * disabled, return whatever we have and log a warning.  Otherwise throw if
    * unable to download a working binary, whether because of network issues or
    * downloads being disabled.
    */
-  public async fetchBinary(restClient: Api): Promise<string> {
+  public async fetchBinary(restClient: Api, label: string | string): Promise<string> {
```

Comment:
Typo? `string | string` collapses down to `string`

Reply:
Whoops good catch lol will fix

You can pass a string OR A STRING!  DOUBLE STRING